resource_customizations/numaplane.numaproj.io/ISBServiceRollout/health.lua

```diff
@@ -2,31 +2,62 @@ local hs = {}
 local healthyCondition = {}
 
 if obj.status ~= nil then
+
+  -- check for certain cases of "Progressing"
+
+  if obj.metadata.generation ~= obj.status.observedGeneration then
+    hs.status = "Progressing"
+    hs.message = "Not yet reconciled"
+    return hs
+  end
+
+  if obj.status.phase == "Pending" then
+    hs.status = "Progressing"
+    hs.message = "Phase=Pending"
+    return hs
+  end
+
+  if obj.status.upgradeInProgress ~= nil and obj.status.upgradeInProgress ~= "" then
+    hs.status = "Progressing"
+    hs.message = "Update in progress"
+    return hs
+  end
+
+  -- now check the Conditions
+
   if obj.status.conditions ~= nil then
     for i, condition in ipairs(obj.status.conditions) do
       if condition.type == "ChildResourcesHealthy" then
         healthyCondition = condition
       end
     end
   end
-
-  if obj.metadata.generation == obj.status.observedGeneration then
-    if (healthyCondition ~= {} and healthyCondition.status == "False" and (obj.metadata.generation == healthyCondition.observedGeneration) and healthyCondition.reason == "ISBSvcFailed") or obj.status.phase == "Failed" then
-      hs.status = "Degraded"
-      if obj.status.phase == "Failed" then
-        hs.message = obj.status.message
-      else
-        hs.message = healthyCondition.message
-      end
-      return hs
-    elseif healthyCondition ~= {} and healthyCondition.status == "True" and (obj.metadata.generation == healthyCondition.observedGeneration) and obj.status.phase == "Deployed" then
-      hs.status = "Healthy"
+  
+  if (healthyCondition ~= {} and healthyCondition.status == "False" and healthyCondition.reason == "ISBSvcFailed") or obj.status.phase == "Failed" then
+    hs.status = "Degraded"
```

Comment:
there's no need to look for `observedGeneration` in the Condition - we already look for it in the overall Status, which tells us if we're still "Progressing"